### PR TITLE
[WIP]added modified_hubber loss

### PR DESCRIPTION
**Pull request by RituRajSingh878**
Added` modified hubber` loss in loss_function to extend the support of loss function.
But I don't understand few things from loss function so anyone can help with this.

`hubber loss` -
https://en.wikipedia.org/wiki/Huber_loss

I will add the test case in later commits.
Thanks

### CODE REVIEW

**Comment by jackgerrits** [comment]
If the PR is not ready, please open it as a draft

File: vowpalwabbit/loss_functions.cc [hunk 1]
```diff
@@ -358,6 +358,58 @@ class poisson_loss : public loss_function
...
+      std::cout << "You are using label " << label << " not -1 or 1 as loss function expects!" << std::endl;
+     float e = max(0, 1-prediction*label);
+    return (prediction*label >= -1.f) ? e*e : -4*label*prediction;
+  }
+
+  float getUpdate(float prediction, float label, float update_scale, float pred_per_update)
+  {
+
+  }
+
```
Comment (RituRajSingh878) [inline vowpalwabbit/loss_functions.cc]: I don't understand how updates work so anyone can explain then I will update it accordingly.

File: vowpalwabbit/loss_functions.cc [hunk 2]
```diff
@@ -358,6 +358,58 @@ class poisson_loss : public loss_function
...
+  float getUnsafeUpdate(float prediction, float label, float update_scale)
+  {
+
+  }
+
```
Comment (RituRajSingh878) [inline vowpalwabbit/loss_functions.cc]: same issue

File: vowpalwabbit/loss_functions.cc [hunk 3]
```diff
@@ -358,6 +358,58 @@ class poisson_loss : public loss_function
...
+  float getRevertingWeight(shared_data* /* sd */, float /* prediction */, float /* eta_t */)
+  {
+
+  }
+
```
Comment (RituRajSingh878) [inline vowpalwabbit/loss_functions.cc]: same issue

**Comment by RituRajSingh878** [comment]
> If the PR is not ready, please open it as a draft

okay, I will close it and open it again as a draft.
Thanks

**Pull request closed without merging**